server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Worker.scala

```diff
@@ -156,14 +158,27 @@ private[deploy] class Worker(
   }
 
   def heartBeatToMaster(): Unit = {
-    val shuffleKeys = new jHashSet[String]
-    shuffleKeys.addAll(partitionLocationInfo.shuffleKeySet)
-    shuffleKeys.addAll(localStorageManager.shuffleKeySet())
-    val response = rssHARetryClient.askSync[HeartbeatResponse](
-      HeartbeatFromWorker(host, rpcPort, pushPort, fetchPort, workerInfo.numSlots,
-        self, shuffleKeys)
-      , classOf[HeartbeatResponse])
-    cleanTaskQueue.put(response.expiredShuffleKeys)
+    try {
+      val shuffleKeys = new jHashSet[String]
+      shuffleKeys.addAll(partitionLocationInfo.shuffleKeySet)
+      shuffleKeys.addAll(localStorageManager.shuffleKeySet())
+      val response = rssHARetryClient.askSync[HeartbeatResponse](
+        HeartbeatFromWorker(host, rpcPort, pushPort, fetchPort, workerInfo.numSlots,
+          self, shuffleKeys)
+        , classOf[HeartbeatResponse])
+      cleanTaskQueue.put(response.expiredShuffleKeys)
+
+      heartBeatFailCnt.getAndSet(0)
+    } catch {
+      case NonFatal(t) =>
+        if (heartBeatFailCnt.addAndGet(1) > 5) {
+          logError(s"Uncaught exception 5 times. Now stopping in thread " +
+              s"${Thread.currentThread().getName}", t)
+          onStop()
+        } else {
+          logError(s"Uncaught exception in thread ${Thread.currentThread().getName}", t)
```

Comment:
Log is not exact, `Uncaught Exception` ?

Reply:
thanks, I have changed the log message.